lib/github/repo_name.rb

```diff
@@ -0,0 +1,24 @@
+module GitHub
+  class RepoName
+    def initialize(client, organization_login, user_login, assignment_slug)
+      @client = client
+      @organization_login = organization_login
+      @user_login = user_login
+      @assignment_slug = assignment_slug
+
+      @suffix_number = 0
+      @suffix_number += 1 while repository?
+    end
+
+    def repository?
+      @client.repository?("#{@organization_login}/#{repo_name}")
+    end
```

Comment:
I would rather this be move over to `GitHub::Repository` as a class method. So that you could do something like

```ruby
GitHub::Repository.present?
```